apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -1253,6 +1268,15 @@ def _write_dataframe_impl(
         return soma_df
 
     soma_df.write(arrow_table)
+
+    # Save the original index name for outgest
+    if original_index_name is None:
+        soma_df.metadata[
+            _DATAFRAME_ORIGINAL_INDEX_NAME
+        ] = _DATAFRAME_ORIGINAL_INDEX_NAME_IS_NONE
```

Comment:
I thought we could put `null`/`None` values into metadata entries. If that’s not true, it might be better to make this value a JSON-encoded value, so `soma_df.metadata[...] = json.dumps(original_index_name)`; then when loading it back up, we don’t have any special-casing; it can just be `json.loads(soma_df.metadata[...])`. Alternately maybe set a special key, like `soma_df.metadata[_..._IS_NONE] = True` (and maybe set `soma_df.metadata[_..._NAME] = ''`) so that there’s not a special value of this key that we have to watch out for.

Reply:
@thetorpedodog thanks.

Whether or not Python `None` writes into metadata, I don't want to know -- I want this language-neutral as much as possible.

I really like the `json.dumps`/`json.loads` approach. :)

Reply:
https://github.com/single-cell-data/TileDB-SOMA/pull/2028/commits/da57874e0ef7ce3ec97e4973a5c9ca64851474f2

Reply:
@thetorpedodog also, experimentally:

```
>>> exp=tiledbsoma.open('foo', 'w')

>>> exp.metadata['hm']=None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/johnkerl/git/single-cell-data/TileDB-SOMA/apis/python/src/tiledbsoma/_tdb_handles.py", line 493, in __setitem__
    _check_metadata_type(key, value)
  File "/Users/johnkerl/git/single-cell-data/TileDB-SOMA/apis/python/src/tiledbsoma/_tdb_handles.py", line 543, in _check_metadata_type
    raise TypeError(f"cannot store {type(obj)} instance as metadata")
TypeError: cannot store <class 'NoneType'> instance as metadata
```